Classes/EventListener/Order/Stock/ProcessOrderCheckStock.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Extcode\CartProducts\EventListener\Order\Stock;
+
+/*
+ * This file is part of the package extcode/cart-products.
+ *
+ * For the full copyright and license information, please read the
+ * LICENSE file that was distributed with this source code.
+ */
+
+use Extcode\Cart\Event\ProcessOrderCheckStockEvent;
+use Extcode\CartProducts\Domain\Model\Product\Product;
+use Extcode\CartProducts\Domain\Repository\Product\ProductRepository;
+use TYPO3\CMS\Core\Messaging\FlashMessage;
+use TYPO3\CMS\Core\Type\ContextualFeedbackSeverity;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+use TYPO3\CMS\Extbase\Utility\LocalizationUtility;
+
+class ProcessOrderCheckStock
```

Comment:
Would suggest to rename the file and class to `CheckStock` analogous to `HandleStock`. The ProcessOrder part should be clear by the namespace `Order\Stock`. 
```suggestion
class CheckStock
```

Reply:
Will happen :) It was just adapted to the event it listens to (`ProcessOrderCheckStockEvent`).